### ISSUE TITLE
Thread worker and progress bar for model loading (and other long running computation)

### ISSUE THREAD

**Issue by constantinpape** [computational-cell-analytics/micro-sam#202]
We should wrap long running computation into a thread worker, so that it does not block the viewer, and that a progress bar / "I am working sign" is shown.

See https://github.com/pyapp-kit/magicgui/pull/579 for how to do that with magicgui.

For now, we could try it where the model is loaded in the 2d annotation tool, https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/sam_annotator/annotator_2d.py#L225 and will eventually refcator it.

**Comment by GenevieveBuckley** [comment]
Here's the example code I [left in a comment on a different thread](https://github.com/computational-cell-analytics/micro-sam/issues/203#issuecomment-1767822848):

```python
from time import sleep
from magicgui import magic_factory
from magicgui.tqdm import tqdm


@magic_factory(
    threshold={"widget_type": "FloatSlider", "max": 20},
    call_button="Calculate!",
)
def example_magic_widget(
    img_layer: "napari.layers.Image",
    threshold: "float"
) -> "napari.types.LabelsData":
    with tqdm() as pbar:
        @thread_worker(connect={"finished": lambda: pbar.progressbar.hide()})
        def long_running():
            sleep(5)

        long_running()
```

**Comment by constantinpape** [comment]
@GenevieveBuckley: You wrote in the other thread 

> Ok, I had a strangely difficult time combining a thread worker with an indeterminate progress bar in the napari plugin widget. Here's what it should look like: ...

Just to clarify: the solution you posted here is now working, correct?

**Comment by constantinpape** [comment]
This is implemented now in #235

**Comment by constantinpape** [comment]
See #263 for follow up
